### PR TITLE
fix merging of api resource data fails or has existing data go missing

### PR DESCRIPTION
**Pull request by esserj**
in case certain xml keys are missing from extending config like properties, item opteration, collection operations or resources

| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 - i know it is a bugfix, but the merger tests are missing from 1.12                  |
| Bug fix?        |yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | /                      |
| License         | MIT                                                          |

Basically what happens is that when you define api extension configuration through app (or multiple bundles)
you have to set at least 1 property in the file, oven though you do not want to make any changes (potentially overriding any future changes to it)
and you have to explicitly set every operation type (itemOperations, collectionOperations, resourceOperations) explictly
like `<itemOperations />` etc or you will end up (based on order of execution) with eg default item operation configuration  from api platform, or subresouce configation goes mising (e.g order override caused order adjustment subresource config from going awol)

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/ApiPlatform/Metadata/MergingXmlExtractor.php [hunk 1]
```diff
@@ -107,8 +107,8 @@ protected function extractPath(string $path): void
                     $resourceMetadata,
                 );
                 $this->properties[$resourceClass] = array_merge(
-                    $this->properties[$resourceClass],
-                    $resourceMetadata['properties'],
+                    $this->properties[$resourceClass] ?? [],
+                    $resourceMetadata['properties'] ?? [],
```
Comment (esserj) [inline src/Sylius/Bundle/ApiBundle/ApiPlatform/Metadata/MergingXmlExtractor.php:111]: null cannot be merged with []
Maybe adjusting \Sylius\Bundle\ApiBundle\ApiPlatform\Metadata\MergingXmlExtractor::buildResource properties to not default to null is better?

File: src/Sylius/Bundle/ApiBundle/ApiPlatform/Metadata/Merger/LegacyResourceMetadataMerger.php [hunk 2]
```diff
@@ -23,6 +23,10 @@ public function merge(array $oldMetadata, array $newMetadata): array
 
         foreach ($newMetadata as $key => $value) {
             if ('properties' === $key) {
+                if ($value === null) {
```
Comment (esserj) [inline src/Sylius/Bundle/ApiBundle/ApiPlatform/Metadata/Merger/LegacyResourceMetadataMerger.php:26]: maybe adjusting \Sylius\Bundle\ApiBundle\ApiPlatform\Metadata\MergingXmlExtractor::buildResource properties is better?

**Comment by esserj** [comment]
this may break api's that accidentally exposed `itemOperations` by not defining them in extending files
as api-platform would have turned them into default operation endpoints (which depending on the endpoint) could have exposed admin only resource modification

**Comment by GSadee** [comment]
Hi @esserj and thank you for your contribution.
Indeed, the current merging behaviour of arrays when the data is not configured in the overwritten file can be confusing and it needs to be improved. Your solution potentially looks OK, although it does change the current behaviour. So, we would need to think through all possible scenarios for overwriting operations/properties/etc and test them in PHPUnit tests accordingly.

**Comment by GSadee** [comment]
I'm closing the PR in favour of https://github.com/Sylius/Sylius/pull/18725
